### PR TITLE
Add CI workflow

### PR DESCRIPTION
**Pull request by clementlgl**

### CODE REVIEW

**Comment by clementlgl** [comment]
Errors must be fixed in another PR.

File: .gitignore
[the remark this answers is not in the document]
Comment (clementlgl) [reply .gitignore:1]: I think it's a good idea to use Zephyr `.gitignore`, even if it's too complete! But it excludes paths like `twister-out*`, which is interesting.
Why place it in the samples folder and not in the root directory?

[the remark this answers is not in the document]
Comment (bperraudin) [reply .gitignore:1]: I think you're right: we should use the default one, at the root of the repository.

File: .github/workflows/driver.yaml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: "Honeywell HPMA115 Sensor Driver CI"
```
[the remark this answers is not in the document]
Comment (bperraudin) [reply .github/workflows/driver.yaml:1]: May be we should name it: "Test driver"

File: samples/sample.yaml [hunk 2]
```diff
@@ -0,0 +1,9 @@
+sample:
+  description: Demonstration of the HPMA115 Dust Sensor driver
```
[the remark this answers is not in the document]
Comment (bperraudin) [reply samples/sample.yaml:2]: OK to remove the description as is done in Zephyr.

And update `name` to: "HPMA115 dust sensor sample"

**Review by bperraudin — CHANGES_REQUESTED** [review]

File: .gitignore
[the remark this answers is not in the document]
Comment (clementlgl) [reply .gitignore:1]: @bperraudin I have seen that the [Zephyr example application](https://github.com/zephyrproject-rtos/example-application.git) uses this [gitignore](https://github.com/zephyrproject-rtos/example-application/blob/main/.gitignore). I think we should use this one, What do you think?

**Review by bperraudin — APPROVED** [review]

**Pull request merged**